src/library/scala/util/Try.scala

```diff
@@ -174,6 +174,29 @@ sealed abstract class Try[+T] {
    *  `s` if this is a `Success`.
    */
   def transform[U](s: T => Try[U], f: Throwable => Try[U]): Try[U]
+
+  /**
+   * Returns `Left` with `Throwable` if this is a `Failure`, otherwise returns `Right` with `Success` value.
+   */
+  def toEither: Either[Throwable, T]
+
+  /**
+   * Applies `fa` if this is a `Failure` or `fb` if this is a `Success`.
+   *
+   * @example {{{
+   * val result: Try[Throwable, Int] = Try { string.toInt }
+   * log(result.fold(
+   *   ex => "Operation failed with " + ex,
+   *   v => "Operation produced value: " + v
+   * ))
+   * }}}
+   *
+   * @param fa the function to apply if this is a `Failure`
+   * @param fb the function to apply if this is a `Success`
+   * @return the results of applying the function
+   */
+  final def fold[U](fa: Throwable => U, fb: T => U): U = toEither.fold(fa, fb)
```

Comment:
I wasn't aware that the choose approach in such cases (like `Try.transform`) is that if an exception is thrown by function applied in on success/completed value, then it fallbacks to the function provided for failure case (BTW it could be good to document it somewhere).